lib/SILGen/SILGenExpr.cpp

```diff
@@ -5090,15 +5090,34 @@ RValue RValueEmitter::emitForceValue(ForceValueExpr *loc, Expr *E,
 void SILGenFunction::emitOpenExistentialExprImpl(
        OpenExistentialExpr *E,
        llvm::function_ref<void(Expr *)> emitSubExpr) {
-  Optional<FormalEvaluationScope> writebackScope;
-
   // Emit the existential value.
   if (E->getExistentialValue()->getType()->is<LValueType>()) {
-    bool inserted = OpaqueValueExprs.insert({E->getOpaqueValue(), E}).second;
+    // Open the existential container right away. We need the dynamic type
+    // to be opened in order to evaluate the subexpression.
+    AccessKind accessKind;
+    if (E->hasLValueAccessKind())
+      accessKind = E->getLValueAccessKind();
```

Comment:
Is this actually ever false?  We're supposed to be asserting that it isn't in the AST verifier.

Reply:
You could have an rvalue `OpenExistentialExpr` that opens an lvalue existential. The AST representation is a bit funky in the lvalue `OpenExistentialExpr` case—it looks like the `OpenExistentialExpr` gets the real access kind, while the access kind on the existential subexpression is always `readwrite`. I wanted to fix this regression before looking more deeply into that. (The previous code apparently disregarded the AST access kind altogether and forwarded it down from the outer lvalue evaluation.)

Reply:
Slava informed me offline that he was originally trying to eliminate access kinds entirely from the AST. Not sure we can do that for OpenExistentialExprs while preserving the intended semantics and the optimization for read-only accesses.